src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+
+    private int threshold = (int) (LOAD_FACTOR * INITIAL_CAPACITY);
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> current : table) {
+            while (current != null) {
+                if (Objects.equals(key, current.key)) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        int index = getTableIndex(table.length, hash);
+        Node<K, V> current = table[index];
+        Node<K, V> newNode = new Node<>(hash, key, value);
+        if (current == null) {
+            table[index] = newNode;
+        } else {
+            for (; ; ) {
+                if (Objects.equals(key, current.key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    break;
+                }
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        if (++size > threshold) {
+            resize();
```

Comment:
No. Our desired logic is that if size passes the threshold(if size is bigger than threshold), we need to resize our map. Let's say our size is 11 and threshold is 12. ++size will result in size becoming 12. 12 is not bigger then 12, so ">=" doesn't match our requirements. On the other hand if we increment our size again, now it will be bigger than the threshold, and that's exactly the moment when we want map resizing to happen